src/ExtractTileOperations.cpp

```diff
@@ -197,14 +369,24 @@ Matmul convert_to_matmul(const Store *op, const string &new_name, AMXOpType op_t
 
     const auto *lhs_load = matches[0].as<Load>();
     const auto *rhs_broadcast = matches[1].as<Broadcast>();
-    if (!lhs_load || !rhs_broadcast) {
+
+    const Cast *rhs_cast = nullptr;
+
+    if (lhs_load && !rhs_broadcast) {
+        // now working on a larger k dimension
+        rhs_cast = matches[1].as<Cast>();
+    } else {
+        rhs_cast = rhs_broadcast->value.as<Cast>();
+    }
+
+    if (!lhs_load || !rhs_cast) {
         return {};
     }
-    const auto *rhs_cast = rhs_broadcast->value.as<Cast>();
+
     if (rhs_cast) {
         if (op_type == AMXOpType::Int8) {
             if (!(rhs_cast->value.type().element_of() == Int(8) || rhs_cast->value.type().element_of() == UInt(8))) {
-                user_assert(false) << "Expected rhs cast of i8/u8";
+                user_error << "Expected rhs cast of i8/u8, got " << rhs_cast->value.type();
```

Comment:
This is likely not enough context for a user_error. Is there any identifying information as to which Func is involved here? Maybe printing the full statement will be more of a clue, along with the message that AMX doesn't support casts to other than i8/u8 on the RHS (if that's accurate).

Reply:
added some more detail to the error here b56aad1eff11e7ec75ba58280e406f568b136949.